worker/upgradeseries/worker.go

```diff
@@ -352,6 +352,14 @@ func (w *upgradeSeriesWorker) pinLeaders() (err error) {
 
 	results, err := w.PinMachineApplications()
 	if err != nil {
+		// If pin machine applications method return not implemented because it's
+		// utilising the legacy leases store, then we should display the warning
+		// in the log and return out. Unpinning leaders should be safe as that
+		// should be considered a no-op
+		if errors.IsNotImplemented(errors.Cause(err)) {
+			w.logger.Warningf("failed to pin machine applications %+v", err)
```

Comment:
Thinking about this, maybe this should be Infof, as it isn't something we want someone to file a bug about/investigate more thoroughly.

Did you see the actual error under this circumstance? do we trust that it will be something a human will understand?

Reply:
I've moved it to Infof.

> do we trust that it will be something a human will understand?

You can see the error message here:

> machine-0: 13:01:20 INFO juju.worker.upgradeseries failed to pin machine applications  unable to get leadership pinner; pinning is not available with the legacy lease manager not implemented

Reply:
I'd rather we play with it a bit to make a bit more sense:

failed to pin machine applications, with legacy lease manager leadership pinning is not implemented

Thoughts?